doc/docs/Mode_Decomposition.md

```diff
@@ -180,61 +176,64 @@ These functions are implemented in [src/dft.cpp](https://github.com/NanoComp/mee
-   \qquad
-   \mathbf{H}^+_{n\parallel} =-\mathbf{H}^-_{n\parallel}
+$$
+\psi^{\text{meep}} =
+ \sum_{n} \left\{   \alpha^+_n \mathbf \psi^+_n(\vec \rho)e^{+i\beta_n x}
+                    + \alpha^-_n \mathbf \psi^-_n(\vec \rho)e^{-i\beta_n x}
+            \right\}
 $$
 
-Taking the inner product of both equations with the $\mathbf{H}$ and $\mathbf{E}$ fields of each eigenmode, we find
-
-$$ \left\langle \mathbf{H}_m \right|\left. \mathbf{E}^{\text{meep}} \right\rangle =+(a_m^+ + a_m^-) v_m A_S $$
+Taking the inner product of this equation with the $\psi_m^\pm$ modes computed
+from MPB (renormalized to unit power), we obtain the mode coefficients:
 
-$$ \left\langle \mathbf{E}_m \right|\left. \mathbf{H}^{\text{meep}} \right\rangle =-(a_m^+ - a_m^+) v_m A_S $$
+$$
+\alpha^\pm_n = \left\langle \psi_m^\pm, \psi^{\text{meep}} \right\rangle \, .
+$$
 
-Thus, by evaluating the integrals on the left-hand side of these equations &mdash; numerically, using the MPB-computed eigenmode fields $\{\mathbf{E}_m, \mathbf{H}_m\}$ and the Meep-computed fields $\{\mathbf{E}^{\text{meep}}, \mathbf{H}^{\text{meep}}\}$ as tabulated on the computational grid &mdash; and combining the results appropriately, we can extract the coefficients $\alpha^\pm_m$. This calculation is carried out by the routine `fields::get_mode_flux_overlap`. Although simple in principle, the implementation is complicated by the fact that, in multi-processor calculations, the Meep fields needed to evaluate the integrals are generally not all present on any one processor, but are instead distributed over multiple processors, requiring some interprocess communication to evaluate the full integral.
+Some simplifications arise from the fact that, in a constant cross-section waveguide, the tangential components of the forward- and backward-traveling propagating modes are related by a mirror reflection in $x$:
+$\mathbf{E}^+_{n\parallel} =+\mathbf{E}^-_{n\parallel}$
+and $\mathbf{H}^+_{n\parallel} =-\mathbf{H}^-_{n\parallel}$.  This means that we only need to compute the MPB modes once for the forward (+) direction.  Also, the $\langle \psi_m^\pm, \psi^{\text{meep}} \rangle$ inner
+product involves four integrals (of terms like $E_y H_z$ etc.) that are computed
+once each and then combined with the correct signs to obtain $\alpha^\pm_n$.
+These integrals are computed numerically by a trapezoidal-type rule on Meep's
+Yee grid, interpolating the MPB-computed eigenmode fields as needed. This calculation is carried out by the routine `fields::get_mode_flux_overlap`. Although simple in principle, the implementation is complicated by the fact that, in multi-processor calculations, the Meep fields needed to evaluate the integrals are generally not all present on any one processor, but are instead distributed over multiple processors, requiring some interprocess communication to evaluate the full integral.
 
+As mentioned above, the Poynting flux $P$ carried by the Meep fields is
```

Comment:
Typo which needs to be corrected: `the Poynting flux $P$ carried by the Meep fields is The Poynting flux carried by the Meep fields...`